js/main.js

```diff
@@ -55,12 +59,45 @@ async function defineCardComponent() {
     //Binds the card's onclick events to flip and show the description popup.
     setupOnClickEvents() {
       this.back.onclick = (event) => this.flipCard();
-      this.front.onclick = (event) => alert("TODO: Description popup.");
+      this.front.onclick = (event) => this.showDetails();
+    }
+
+    showDetails() {
+      showDetailsDialog(this.data, this.getImageURL());
     }
   }
   customElements.define("tcg-card", Card);
 }
 
+function showDetailsDialog(data, cardUrl) {
+  DETAILS_DIALOG.classList.remove("hidden");
+  DETAILS_DIALOG.getElementsByClassName("dialog-title")[0].innerHTML = "";
+  DETAILS_DIALOG.getElementsByClassName("dialog-title")[0].insertAdjacentHTML(
+    "beforeend",
+    `
+    <h2 class="card-name">${data["Card Display Name"]}</h2>
+    <p>${data["Card Level"]} | ${data["Card HP"]}HP |  ${data["Card Element"]} |  #${data["Collector Number"]}</p>
+    <p>Artist: ${data["Artist Credit"]} | Writer: ${data["Writer Credit"]}</p>
+  `
+  );
+  DETAILS_DIALOG.getElementsByClassName("details-dialog-card")[0].src = cardUrl;
+  DETAILS_DIALOG.getElementsByClassName("details-dialog-text")[0].innerHTML =
+    "";
+  DETAILS_DIALOG.getElementsByClassName(
+    "details-dialog-text"
+  )[0].insertAdjacentHTML(
+    "beforeend",
+    `
+    <h3>${data["Attack 1 Name"]}</h3>
+    <p>${data["Attack 1 Description"]}</p>
+    <p><i>${data["Attack 1 Extended Lore"]}</i></p>
+    <h3>${data["Attack 2 Name"]}</h3>
+    <p>${data["Attack 2 Description"]}</p>
+    <p><i>${data["Attack 2 Extended Lore"]}</i></p>
```

Comment:
I'd prefer to keep what I have since it works, and to make it part of an HTML template I'd presume I have to do
```
      this.front = this.getElementsByClassName("card-front")[0];
      this.back = this.getElementsByClassName("card-back")[0];
```
except for all 10 card attributes, which sounds tedious orz